be/src/exprs/in_predicate.cpp

```diff
@@ -35,13 +33,17 @@ InPredicate::InPredicate(const TExprNode& node)
           _null_in_set(false),
           _hybrid_set() {}
 
-InPredicate::~InPredicate() {}
+InPredicate::~InPredicate() {
+    if (_should_delete) {
```

Comment:
really we need `should_delete` ? use  _hybrid_set == nullptr or just use unique_ptr?

Reply:
InPredicate have another prepare function `InPredicate::prepare(RuntimeState* state, HybridSetBase* hset)`, when this prepare be used, we should do not delete `_hybrid_set`